pallets/parachain-staking/src/lib.rs

```diff
@@ -1902,14 +2053,20 @@ pub mod pallet {
 					}
 					(false, true) => {
 						// candidate pushed out the least staked collator which is now at position
-						// min(max_selected_top_candidates, top_candidates - 1) in TopCandidates
-						let old_col_idx = max_selected_candidates.min(top_candidates.len().saturating_sub(1));
+						let (drop_self, drop_delegators) = match max_selected_candidates.cmp(&top_candidates.len()) {
```

Comment:
Why these changes here?

Reply:
This is a bug, which the sanity tests revealed. If the top candidates are not full, we have not to decrease the total stake.